packages/react-components/react-table/src/components/DataGridRow/useDataGridRow.tsx

```diff
@@ -56,12 +58,13 @@ export const useDataGridRow_unstable = (props: DataGridRowProps, ref: React.Ref<
     {
       appearance,
       'aria-selected': selectable ? selected : undefined,
+      tabIndex: tabbable && !isHeader ? 0 : undefined,
+      ...(focusMode === 'composite' && !isHeader && compositeRowTabsterAttribute),
```

Comment:
this header logic is starting to be a bit crazy, should refactor it in the future https://github.com/microsoft/fluentui/issues/28398